process/Style-guide.md

```diff
@@ -0,0 +1,250 @@
+
+
+## Indentation
+
+* Indent your code blocks with 2 spaces (not 4 as in the style guide).
+* In the case of a hanging indent, use four spaces (according to the style guide).
+
+## Naming
+
+* Use full English words everywhere. For example, use Event not Evt and Description not Desc.Python-3-Guide.md
+* Acronyms and initialisms should be preserved, such as HTMLParser and not HtmlParser.
+* Use "cls" as the name of the class variable in preference to "klass"
+* Method and function names follow the following logic (overriding the Google Python Style Guide):
+
+
+Type | Public | Internal
+---- | ---- | ----
+Functions | **CapWords()** | **_CapWords()** (protected) and **__CapWords()** (private)
+
+
+## Linting
+
+log2timeline uses pylint 1.7.x to enforce some additional best practices to keep the source code more readable. These are:
+
+* Limit the maximum number of arguments for function or method to 10
+
+Other things to keep in mind when using Pylint:
+* Use textual pylint overrides e.g. "# pylint: disable=no-self-argument" instead of "# pylint: disable=E0213". For a list of overrides see: http://docs.pylint.org/features.html
+* The canonical pylint configuration for log2timeline project is generated by l2tdevtools. 
+
+
+### Function and method arguments
+
+* Prefix unused function or method arguments with `unused_`. 
+* In addition to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html#Comments) sort keyword arguments alphabetically by name.
+* Call keyword argument with their keyword prefix, not as positional arguments. For example:
+```python
+
+def SomeMethod(self, alternate=False, secondary=True):
+  pass
+
+# No:
+SomeMethod(True, False)
+
+# Yes:
+SomeMethod(secondary=False)
+```
+
+## Strings
+
+* Quote strings as ' or """ and not "
+  * Quote strings in command line arguments (argparse) as "
+* Textual strings should be Unicode strings.
+  * Use the [unicode_literals](http://python-future.org/unicode_literals.html) module to make all strings unicode by default.
+* Use the format() function instead of the %-style of formatting strings.
+  * Use positional or parameter format specifiers with typing e.g. '{0:s}' or '{text:s}' instead of '{0}', '{}' or '{:s}'. If we ever want to have language specific output strings we don't need to change the entire codebase (again). It also makes is easier in determining what type every parameter is expected to be.
+
+## Exceptions
+
+* When catching exceptions use "as exception:" not some alternative form like "as error:" or "as details:"
+* Raise exceptions like this: ```raise MyException('Error message')``` or ```raise MyException```.
+* Although Python allows for ```try ... except ... else``` we prefer not to use it.
+* Make exception messages as useful and descriptive and possible. For example, if an argument is out of an acceptable range, print the invalid value to speed-up debugging.
+
+##Return statements
```

Comment:
Change to `## Return statements` (add space)

Reply:
You'll need to add a space here otherwise markdown is broken.

Reply:
Woops - fixed.